### PR TITLE
Push Notification Support

### PR DESCRIPTION
**Pull request by sojanpr**
App will check for extra value for key  “branch” in intent. If
available this value is  send with the install/open api as  `
"push_identifier": “my_branch_url”,`
Clients need  to add the branch url as “branch” in their extra
values for push notification

@aaustin @Sarkar @EvangelosG

### CODE REVIEW

File: Branch-SDK/src/io/branch/referral/Branch.java [hunk 1]
```diff
@@ -1289,6 +1289,37 @@ private void executeClose() {
     }
 
     private boolean readAndStripParam(Uri data, Activity activity) {
+
+        // Capture the intent URI and extra for analytics in case started by external intents such as  google app search
+        try {
```
Comment (aaustin) [inline Branch-SDK/src/io/branch/referral/Branch.java]: this is the change from the other PR merged in, correct?

**Comment by aaustin** [comment]
If you confirm the above :+1:

File: Branch-SDK/src/io/branch/referral/Branch.java (hunk 1 above)
Comment (sojanpr) [inline Branch-SDK/src/io/branch/referral/Branch.java]: Yes this Branch is cut from the branch for "Capture_External_URI" fix Branch

**Comment by sojanpr** [comment]
Merged

**Pull request closed without merging**